### ISSUE TITLE
Can't build with llvm 3.6.2 on OS X

### ISSUE THREAD

**Issue by iMichka** [CastXML/CastXML#40]
Hi

I am trying to work on CastXML's packaging; so that we can provide binaries through homebrew on Mac.
I am on OS X 10.11.1 (El Capitan) and installed llvm through homebrew (brew install llvm --with-clang).
The install tree is in /usr/local/Cellar/llvm/.

I tested with the corresponding clang version and with the clang version provided by Apple. I get the following linker error:

```
Undefined symbols for architecture x86_64:
  "std::bad_cast::bad_cast()", referenced from:
      std::__1::basic_filebuf<char, std::__1::char_traits<char> >::seekoff(long long, std::__1::ios_base::seekdir, unsigned int) in Utils.cxx.o
      std::__1::basic_filebuf<char, std::__1::char_traits<char> >::sync() in Utils.cxx.o
      std::__1::basic_filebuf<char, std::__1::char_traits<char> >::underflow() in Utils.cxx.o
      std::__1::basic_filebuf<char, std::__1::char_traits<char> >::overflow(int) in Utils.cxx.o
  "std::bad_cast::~bad_cast()", referenced from:
      std::__1::basic_filebuf<char, std::__1::char_traits<char> >::seekoff(long long, std::__1::ios_base::seekdir, unsigned int) in Utils.cxx.o
      std::__1::basic_filebuf<char, std::__1::char_traits<char> >::sync() in Utils.cxx.o
      std::__1::basic_filebuf<char, std::__1::char_traits<char> >::underflow() in Utils.cxx.o
      std::__1::basic_filebuf<char, std::__1::char_traits<char> >::overflow(int) in Utils.cxx.o
  "std::terminate()", referenced from:
      ___clang_call_terminate in castxml.cxx.o
      ___clang_call_terminate in Detect.cxx.o
      ___clang_call_terminate in Output.cxx.o
      ___clang_call_terminate in RunClang.cxx.o
      ___clang_call_terminate in Utils.cxx.o
ld: symbol(s) not found for architecture x86_64
```

Cmake was run with the following options:
>  cmake DCMAKE_CXX_COMPILER=/usr/local/Cellar/llvm/3.6.2/bin/clang++ -DLLVM_DIR=/usr/local/Cellar/llvm/3.6.2/share/llvm/cmake

Here is cmake's output:
https://gist.github.com/iMichka/9e256769e3bbd691b886

And here make:
https://gist.github.com/iMichka/32592928dc466be6ff86

Maybe you can help me out here. It would be great to have a reliable way to compile CastXML on OS X (and I would prefer not to keep multiple llvm versions around on my macbook, this is eating up a lot of space :)).

**Comment by bradking** [comment]
> cmake -DCMAKE_CXX_COMPILER=/usr/local/Cellar/llvm/3.6.2/bin/clang++ -DLLVM_DIR=/usr/local/Cellar/llvm/3.6.2/share/llvm/cmake

You don't need to compile CastXML with the Clang that it is linking against.  You should compile CastXML using the same compiler that was used to build LLVM/Clang (i.e. the compiler that brew used to build it).

**Comment by iMichka** [comment]
Yes, I also tries that (by not passing the CMAKE_CXX_COMPILER flag). Cmake will pick up OS X's clang++, which is the same that built homebrew's llvm.

I will perhaps build a 3.6.2 version myself. I had a 3.6.0 version which I built myself and it worked with that one.

**Comment by bradking** [comment]
I ran `brew install llvm --with-clang` and was able to reproduce this.  The problem is that CastXML is not building with `-fno-exceptions` as LLVM/Clang did.  Fixed by commit acd58dac6f950a4c6eece2b05f7ae5d7467efe72.

**Comment by iMichka** [comment]
Thanks for having looked into this. Works perfectly. We will soon have castxml binaries on homebrew.
